src/nccl_ofi_rdma.cpp

```diff
@@ -2905,7 +2908,13 @@ static inline int allocate_rdma_recv_req(
 	* wrap around works fine too */
 	uint16_t slot = req->msg_seq_num % NCCL_OFI_CTRL_MAILBOX_SIZE;
 	r_comm->ctrl_mailbox[slot].msg_seq_num = req->msg_seq_num & MSG_SEQ_NUM_MASK;
-	r_comm->ctrl_mailbox[slot].buff_addr = (uint64_t)buff;
+	/* Populate buffer address/offset based on whether provider supports FI_MR_VIRT_ADDR */
+	if (virt_addr_mr) {
```

Comment:
per above, let's get rid of this.  The conditional is going to be worse than the math, perf wise.